apps/emqx/src/emqx_channel.erl

```diff
@@ -2623,6 +2624,18 @@ do_check_sub_caps(ClientInfo, [TopicFilter = {{Topic, SubOpts}, ?RC_SUCCESS} | M
     case emqx_mqtt_caps:check_sub(ClientInfo, Topic, SubOpts) of
         ok ->
             do_check_sub_caps(ClientInfo, More, [TopicFilter | Acc]);
+        {ok, MaxQoS} ->
+            ?SLOG(
+                warning,
```

Comment:
maybe `info` level otherwise it can be excessive logging if large number of clients behave the same way.